tests/test_BT_piecewise.py

```diff
@@ -0,0 +1,560 @@
+        )
+
+
+def add_groups_and_make_toas(build_model, build_piecewise_model_with_two_pieces, param):
+    # function to build the models for specific edge cases
+    spacing = (build_model.FINISH.value - build_model.START.value) / 2
+    if param == "non-overlapping complete group coverage":
+        model = build_piecewise_model_with_two_pieces
+        model.setup()
+        # print(model.as_parfile())
+        toas = make_generic_toas(model)
+        return model, toas
+    elif param == "overlapping groups":
+        model2 = build_model
+        model2.remove_range(0)
+        model2.add_group_range(
+            model2.START.value - 1, model2.START.value + spacing + 100, j=0
+        )
+        model2.add_group_range(
+            model2.START.value + spacing - 100, model2.FINISH.value + 1, j=1
+        )
+        model2.add_piecewise_param("T0", "d", model2.T0.value + 1e-5, 0)
+        model2.add_piecewise_param("T0", "d", model2.T0.value + 2e-5, 1)
+        model2.add_piecewise_param("A1", "ls", model2.A1.value + 1e-5, 0)
+        model2.add_piecewise_param("A1", "ls", model2.A1.value + 2e-5, 1)
+        model2.setup()
+        toas = make_generic_toas(model2)
+        return model2, toas
+    elif param == "non-complete group coverage":
+        model3 = build_model
+        model3.remove_range(0)
+        model3.add_group_range(
+            model3.START.value + spacing, model3.FINISH.value + 1, j=0
+        )
+        model3.add_piecewise_param("T0", "d", model3.T0.value + 1e-5, 0)
+        model3.setup()
+        toas = make_generic_toas(model3)
+        return model3, toas
+
+
+def make_generic_toas(model):
+    # makes toas to go with the edge cases
+    return pint.simulation.make_fake_toas_uniform(
+        model.START.value, model.FINISH.value - 1, 20, model
+    )
+
+
+def convert_int_into_index(i):
+    # converts i to 4-digit integer: 1->0001, 1010->1010
+    return f"{int(i):04d}"
+
+
+def check_if_in_piece(model, toas):
+    # Tests if the in_piece fn works. in_piece is used in the fitter to only apply the adjustment to toas inside a group. Returns truth array, true if toa lies in the T0X_i we are interested in and false elsewhere
+    # Check to see if these are being allocated correctly.
+    is_toa_in_each_group = []
+    for i in range(model.get_number_of_groups()):
+        par = f"T0X_{convert_int_into_index(i)}"
+        is_toa_in_each_group.append(
+            model.does_toa_reference_piecewise_parameter(toas, par)
+        )
+    return is_toa_in_each_group
+
+
+def return_truth_array_based_on_group_boundaries(model, barycentric_toa):
+    boundaries = model.get_group_boundaries()
+    upper_edge_of_lower_group = boundaries[1][0]
+    lower_edge_of_upper_group = boundaries[0][1]
+    truth_array_comparison = [
+        [barycentric_toa.value <= upper_edge_of_lower_group],
+        [barycentric_toa.value >= lower_edge_of_upper_group],
+    ]
+    return truth_array_comparison
+
+
+@pytest.mark.parametrize(
```

Comment:
You can pass a list of functions to `parametrize`.

Reply:
Didn't know that, thanks!